yeat/cli/auto_pop.py

```diff
@@ -0,0 +1,112 @@
+# -------------------------------------------------------------------------------------------------
+# Copyright (c) 2024, DHS. This file is part of YEAT: http://github.com/bioforensics/yeat
+#
+# This software was prepared for the Department of Homeland Security (DHS) by the Battelle National
+# Biodefense Institute, LLC (BNBI) as part of contract HSHQDC-15-C-00064 to manage and operate the
+# National Biodefense Analysis and Countermeasures Center (NBACC), a Federally Funded Research and
+# Development Center.
+# -------------------------------------------------------------------------------------------------
+
+from collections import defaultdict
+import json
+from pathlib import Path
+
+
+EXTENSIONS = (".fastq", ".fastq.gz", ".fq", ".fq.gz")
+
+
+class AutoPopError(ValueError):
+    pass
+
+
+class AutoPop:
+    def __init__(self, args):
+        self.samples = self.get_samples(args.samples)
+        self.check_samples()
+        self.check_seq_path(args.seq_path)
```

Comment:
The `args` argument here hides all the data that the `AutoPop` object depends on to work properly. If we explicitly pass that data in discrete arguments, it will more clearly communicate what the object needs and what the code is doing.

It looks like there are four arguments: samples, files, seq_path, and outfile. "Clean Code" purists would argue that this is too many, and in some cases they may be right. But I'm not sure I see a way to implement this feature without passing at least the first three to the same object. You could definitely simplify by moving the `write_config_file` method outside of the constructor, requiring the user to call it explicitly, and passing `outfile` as an argument to that method.